lib/features/children/generosity_challenge/pages/generosity_challenge_introduction.dart

```diff
@@ -34,9 +35,13 @@ class _GenerosityChallengeIntruductionState
     final challenge = context.read<GenerosityChallengeCubit>();
     const pictureHeight = 150.0;
     return Scaffold(
-      appBar: const GenerosityAppBar(
+      appBar: GenerosityAppBar(
         title: 'Generosity Challenge',
-        leading: null,
+        leading: GenerosityBackButton(
+          onPressed: () => context.goNamed(
```

Comment:
Is this needed if we already change the navigation stack with pushReplacementNamed?